code/edgeidentity/src/test/java/com/adobe/marketing/mobile/edge/identity/IdentityStateTests.java

```diff
@@ -542,4 +559,295 @@ public void testUpdateLegacyExperienceCloudId_notSetWhenExistingIsNull() {
 		assertNull(state.getIdentityProperties().getECIDSecondary());
 		verify(mockSharedPreferenceEditor, Mockito.times(0)).apply();
 	}
+
+	// ======================================================================================================================
+	// Tests for method : updateAdvertisingIdentifier(final Event event, final SharedStateCallback callback)
+	// ======================================================================================================================
+
+	// With consent change
+	@Test
+	public void testupdateAdvertisingIdentifier_notSetWhenExistingIsNull() {
+		IdentityState state = new IdentityState(new IdentityProperties());
+		state.getIdentityProperties().setECID(new ECID());
+
+		state.updateLegacyExperienceCloudId(null);
+
+		assertNull(state.getIdentityProperties().getECIDSecondary());
+		verify(mockSharedPreferenceEditor, Mockito.times(0)).apply();
+	}
+
+	@Test
+	public void testUpdateAdvertisingIdentifier_whenNull_thenChangedToValid() throws Exception {
+		assertUpdateAdvertisingIdentifier_updatedWithConsentChange(null, "adID", "adID", "y");
+	}
+
+	@Test
+	public void testUpdateAdvertisingIdentifier_whenEmpty_thenChangedToValid() throws Exception {
+		assertUpdateAdvertisingIdentifier_updatedWithConsentChange("", "adID", "adID", "y");
+	}
+
+	@Test
+	public void testUpdateAdvertisingIdentifier_whenValid_thenChangedToEmpty() throws Exception {
+		assertUpdateAdvertisingIdentifier_updatedWithConsentChange("oldAdId", "", null, "n");
+	}
+
+	@Test
+	public void testUpdateAdvertisingIdentifier_whenValid_thenChangedToAllZeros() throws Exception {
+		assertUpdateAdvertisingIdentifier_updatedWithConsentChange(
+			"oldAdId",
+			IdentityConstants.Default.ZERO_ADVERTISING_ID,
+			null,
+			"n"
+		);
+	}
+
+	@Test
+	public void testUpdateAdvertisingIdentifier_whenAllZeros_thenChangedToEmpty() throws Exception {
+		assertUpdateAdvertisingIdentifier_updatedWithConsentChange(
+			"oldAdId",
+			IdentityConstants.Default.ZERO_ADVERTISING_ID,
+			null,
+			"n"
+		);
+	}
+
+	@Test
+	public void testUpdateAdvertisingIdentifier_whenAllZeros_thenSameValue() throws Exception {
+		assertUpdateAdvertisingIdentifier_updatedWithConsentChange(
+			IdentityConstants.Default.ZERO_ADVERTISING_ID,
+			IdentityConstants.Default.ZERO_ADVERTISING_ID,
+			null,
+			"n"
```

Comment:
Got it. This use-case to not be possible since the extension rejects adid sycs today so it cannot have an all zeroes stored prior to this feature being added. 
I would suggest we remove this one and we rather cover this use-case with [functional tests](https://github.com/adobe/aepsdk-edgeidentity-android/tree/main/code/edgeidentity/src/androidTest/java/com/adobe/marketing/mobile/edge/identity). We can then have tests like when calling MobileCore.setAdvertisingIdentifier(ALL_ZEROES) twice there is no consent event dispatched.

Reply:
That makes sense, I've removed this test case and added a functional test case for sending all zeros twice